### DISCUSSION TITLE
How works and how to use continuation of run?

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#480] [Q&A] [answered]
Doc has little to nothing...
What is ideal strategy for not to lose long simulation due to crash and adjust something, restart the code from the recent way-point and continue run? In some other PIC codes all outputs are restartable. Not with Epoch?

**Comment by TomGoffrey** [comment, marked as answer]
Not all outputs are restartable, no.

Restarting from a snapshot can be achieved using the control block. See here: https://epochpic.github.io/documentation/input_deck/input_deck_control.html

You can control if outputs are restartable via the output block. See here:
https://epochpic.github.io/documentation/input_deck/input_deck_output_block.html

I'm not sure what you think is missing from the documentation?

As for the ideal strategy - that's up to you as a user. For a large simulation writing a restartable snapshot can require non-negligible amounts of time (hence it's optional) - so you'll have to decide how often to do it. It's very much dependent on the problem and system you're running EPOCH on, so I can't really advise further.

**Comment by DanRRRR** [comment]
> I'm not sure what you think is missing from the documentation?
 

Thanks for the references, there i was able to get the info finally (though ideally i'd like everybody preferred also to have couple real examples there showing how specifically to do that in simplest cases). May be would be good to devote a separate paragraph to that topic. 

Why i did not find the needed info is because  when you type "Restart" in the website's  the website gives you 44 topics and mostly NONE of them having any exact content or even having the  word "Restart" there. Here are topics it show in the order user sees: **Quick Start** - nothing there, not a word, nothing about restart, **Overview** - no restart or anything related, **Workshop examples** - not a word, **IDL or GDL** - totally irrelevant too  etcetcetc. And at the end you give up to search further :)

Also the question is:  if i will restart with different amount of processors - will it work ?

**Comment by DanRRRR** [comment]
This documentation need more examples, 3x more, or even 10x more to be usable for novices.  

A lot of info on new style, old style... multiple blocks...rolling dumps...what's included and not included to save the space... the whole day reading... but how actually to do the restart for continuation  - not a  single word or an example. Am i reading wrong documentation? 

Funny, do you know how done the continuation in one  pretty well known PIC code? You just  add few symbols to the usual call line. In our case the whole way of continuation and the entire documentation about continuation would look like this 

echo Data | mpirun -np 1000 ./bin/epoch3d --init=YourLastFile

"Add the  --ini=YourLastFile to your terminal or batch call" is an entire documentation on continuation 

 Continuation needs simplified version when you do not need to read all of that because over the last decade price per unit storage decreased an order of magnitude. Just the NVMe drives decreased in price 4 times  and increased 4x in speed during last 3 years. Really, for me just simpler  to rerun the EPOCH code because it is so fast than to find where the heck to find how continuation actually done in it  :)

Or may be someone will show here an example here how to do simplest, more complex and rolling continuation on  cone.deck demo ?
